doc/move_group_interface/src/move_group_interface_tutorial.cpp

```diff
@@ -32,169 +32,481 @@
+    move_group_.setPoseTarget(pose);
+    // Planning to the goal, and then moving is identical to the joint space goal.
+    //
+    // END_SUB_TUTORIAL
+    bool success = (move_group_.plan(plan) == moveit::planning_interface::MoveItErrorCode::SUCCESS);
+
+    ROS_INFO_NAMED("tutorial", "Visualizing plan (pose space goal) %s", success ? "" : "FAILED");
+    return success;
+  }
+
+  /**
+  * Adds the given planning constraints
+  **/
+  void addConstraint(moveit_msgs::Constraints test_constraints)
+  {
+    // BEGIN_SUB_TUTORIAL add_constraint
+    // Add it to move_group
+    move_group_.setPathConstraints(test_constraints);
+    // Planning with constraints can be slow because every sample must call an inverse kinematics
+    // solver. Lets increase the planning time from the default 5 seconds to be sure the planner
+    // has enough time to succeed.
+    move_group_.setPlanningTime(10.0);
+    // END_SUB_TUTORIAL
+  }
+
+  /**
+  * Removes planning constraints
+  **/
+  void removeConstraints()
+  {
+    // BEGIN_SUB_TUTORIAL remove_constraint
+    // After moving it is often useful to revert back to the default settings.
+    move_group_.clearPathConstraints();
+    move_group_.setPlanningTime(5.0);
+    // The demo should show two paths to compare constrained plan with an unconstrained plan.
+    // END_SUB_TUTORIAL
+  }
+
+  /**
+  * Populates the plan with a cartesian path following the given points in order
+  **/
+  double planCartesianPath(std::vector<geometry_msgs::Pose> target_poses, moveit::planning_interface::MoveGroupInterface::Plan& plan)
+  {
+    // BEGIN_SUB_TUTORIAL cartesian_plan
+    //
+    // Cartesian motions are frequently needed to be slower for actions such as approach and
+    // retreat grasp motions. Here we demonstrate how to reduce the speed of the robot arm via
+    // a scaling factor of the maxiumum speed of each joint. Note this is not the speed of the
+    // end effector point.
+    move_group_.setMaxVelocityScalingFactor(0.1);
+
+    // We want the Cartesian path to be interpolated at a resolution of 1 cm
+    // which is why we will specify 0.01 as the max step in Cartesian
+    // translation.  We will specify the jump threshold as 0.0, effectively disabling it.
+    // Warning - disabling the jump threshold while operating real hardware can cause
+    // large unpredictable motions of redundant joints and could be a safety issue.
+    moveit_msgs::RobotTrajectory trajectory;
+    const double jump_threshold = 0.0;
+    const double eef_step = 0.01;
+    double fraction = move_group_.computeCartesianPath(target_poses, eef_step, jump_threshold, trajectory);
+
+    // While we didn't use the planner to create this trajectory it still can be added to a plan
+    plan.trajectory_ = trajectory;
+    // Moving the robot with this plan is the same as before.
+    //
+    // When finished remember to reset the scalingfactor to return to normal speeds.
+    move_group_.setMaxVelocityScalingFactor(1);
```

Comment:
nit: If the user set this elseware, this could trip up their code. I would maybe copy the current factor to a local variable and then use it here.

Reply:
MoveGroup only has a setter for this and no getter